packages/plugin-legacy/src/index.ts

```diff
@@ -519,6 +566,16 @@ function viteLegacyPlugin(options: Options = {}): Plugin[] {
         }
       }
 
```

Comment:
I tried simply adding an if statement at the end of `detectModernBrowserDetector`, but it doesn't seem to have worked 👀
It seems that `__vite_legacy_guard` inside `modernChunkLegacyGuard` is not being executed from anywhere. 
Could you explain how it works?

Reply:
> It seems that `__vite_legacy_guard` inside `modernChunkLegacyGuard` is not being executed from anywhere.
> Could you explain how it works?

The current `modernChunkLegacyGuard` uses the fact that browsers errors without executing a script if the script has invalid syntax (unsupported syntax). But in this case, we are judging with runtime code instead of a syntax. So I was suggesting adding the code in `modernChunkLegacyGuard` (i.e. changing `modernChunkLegacyGuard` to `` `if(HTMLScriptElement&&HTMLScriptElement.supports("importmap"))throw new Error;`);export function __vite_legacy_guard(){${detectModernBrowserDetector}};` ``.

That said, I noticed that the script that has the `throw new Error` won't run but still the scripts imported by that script would run. Hmm...